components/inputs/input-date.js

```diff
@@ -226,16 +226,22 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		// open dropdown on down arrow or enter and focus on calendar focus date
 		if (e.keyCode === 40 || e.keyCode === 13) {
 			this._dropdown.open();
-			this._calendar.focus();
+			requestAnimationFrame(async() => {
```

Comment:
otherwise focus goes to the wrong date (it will focus on the date where the previous focus date was in the previously shown month)

Reply:
I'm slightly confused... it's listening to the `change` event on `d2l-input-text`, which is only going to fire once the input loses focus. But hitting the down key (or ENTER key) isn't going to trigger a focus loss, so how does waiting for an animation frame solve this? Maybe opening the dropdown causes focus loss?

Reply:
I think the enter key is triggering the `change` event (though that's made me realize that this doesn't work for the down arrow)

Reply:
I see. Yeah seems like we need to treat the ENTER and down arrow keys the same way as blurring the input -- trigger a parse & validate, which may result in the previous value coming back. Only after all that happens should the dropdown open and focus be applied.